linker/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -846,17 +858,19 @@ private final class IRChecker(unit: LinkingUnit,
           }
         }
 
-      case Apply(receiver, Ident(method, _), args) =>
+      case Apply(flags, receiver, Ident(method, _), args) =>
+        if (flags.isPrivate)
+          reportError(s"Illegal flag for Apply: Private")
```

Comment:
Because flags can be generalized, for example for #1961. I should probably have mentioned that upfront, but I thought I'd later reuse those `flags` for the call-site inline/noinline hints.

Reply:
Ok, that's fair. Let's keep it that way then.